front/components/agent_builder/capabilities/shared/TimeFrameSection.tsx

```diff
@@ -47,20 +47,15 @@ interface TimeFrameSectionProps {
 
 export function TimeFrameSection({ actionType }: TimeFrameSectionProps) {
   const { setValue, getValues } = useFormContext();
-  const [isChecked, setIsChecked] = useState(() => !!getValues("timeFrame"));
-
-  const { field: durationField } = useController<
-    CapabilityFormData,
-    "timeFrame.duration"
-  >({
-    name: "timeFrame.duration",
-  });
+  const [isChecked, setIsChecked] = useState(
+    () => !!getValues("configuration.timeFrame")
+  );
 
-  const { field: unitField } = useController<
+  const { field: timeFrameField } = useController<
```

Comment:
I need to only have timeFrameField, seems like when you use useController it create value with null automatically so instead of 
```
timeframe: null
``` 
it becomes 
```
timeframe: { duration: null, unit: null}
```

which was a type nightmare because we don't expect these fields to be null